pybossa/util.py

```diff
@@ -1225,21 +1225,22 @@ def admin_or_project_owner(user, project):
 def process_tp_components(tp_code, user_response):
     """grab the 'pyb-answer' value and use it as a key to retrieve the response
     from user_response(a dict). The response data is then used to set the
-    'initial-value' or ':initial-value' based on different components"""
+    :initial-value' """
     soup = BeautifulSoup(tp_code, 'html.parser')
+
+    # Disable autosave so that response for different users will be different
+    task_presenter_elements = soup.find_all("task-presenter")
+    for task_presenter in task_presenter_elements:
+        remove_element_attributes(task_presenter, "auto-save-seconds")
+        remove_element_attributes(task_presenter, "allow-save-work")
```

Comment:
could this be related to the task presenter mode from [here](https://github.com/bloomberg/pybossa/blob/a961052662fc230808bd3937be1037397e997977/pybossa/view/projects.py#L1203); modes such as view-only or when checking each individual answers?

Reply:
No matter which mode we are using (cherry-pick, read-only and etc), we read the TP code from the DB. If the TP code has attributes like "auto-save-seconds" or "allow-save-work", the user response will be saved to browser's IndexedDB in some interval. When we open a TP code with user response, then user's response is saved to IndexedDB as a key value pair (key is task ID and value is the user response). Next time, reading the same task, the response will be retried from the Brower's IndexeDB according to the task ID.

Reply:
Correct me if wrong, we do not want TP code to read from Browser's IndexDB only with read-only(opening tasks w/o obtaining locks) & view-only(viewing task answers) modes?

Reply:
Thanks for explaining over chat. I was under impression that operation under task_presenter [here](https://github.com/bloomberg/pybossa/blob/a961052662fc230808bd3937be1037397e997977/pybossa/view/projects.py#L1202) would be governed by `mode`. However, it looks like task_submitter_id is passed to this function only when task answers are retrieved which later controls removal of auto-save attributes. 
Can we add comments [here](https://github.com/bloomberg/pybossa/blob/a961052662fc230808bd3937be1037397e997977/pybossa/view/projects.py#L1202) explaining `task_submitter_id` would be passed only when viewing the task answers that will disable auto-save attributes

Reply:
Actually, both TP code only needs to be processed with viewing completed response. So both "process_tp_components" and "process_table_component" are only called for viewing the answer.

Reply:
Thanks for adding docstring